### PR TITLE
[vcpkg] #11129: Add VCPKG_LINKER_FLAGS_<CONFIG>.

### PR DESCRIPTION
**Pull request by psfrolov**
Resolves issue #11129.

### CODE REVIEW

File: scripts/toolchains/windows.cmake [hunk 1]
```diff
@@ -22,15 +22,17 @@ if(NOT _CMAKE_IN_TRY_COMPILE)
...
     set(CMAKE_C_FLAGS_DEBUG "/D_DEBUG ${VCPKG_CRT_LINK_FLAG_PREFIX}d /Z7 /Ob0 /Od /RTC1 ${VCPKG_C_FLAGS_DEBUG}" CACHE STRING "")
     set(CMAKE_CXX_FLAGS_RELEASE "${VCPKG_CRT_LINK_FLAG_PREFIX} /O2 /Oi /Gy /DNDEBUG /Z7 ${VCPKG_CXX_FLAGS_RELEASE}" CACHE STRING "")
     set(CMAKE_C_FLAGS_RELEASE "${VCPKG_CRT_LINK_FLAG_PREFIX} /O2 /Oi /Gy /DNDEBUG /Z7 ${VCPKG_C_FLAGS_RELEASE}" CACHE STRING "")
 
-    set(CMAKE_SHARED_LINKER_FLAGS_RELEASE "/DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF ${VCPKG_LINKER_FLAGS}" CACHE STRING "")
-    set(CMAKE_EXE_LINKER_FLAGS_RELEASE "/DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF ${VCPKG_LINKER_FLAGS}" CACHE STRING "")
+    set(CMAKE_SHARED_LINKER_FLAGS_RELEASE "/DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF ${VCPKG_LINKER_FLAGS} ${VCPKG_LINKER_FLAGS_RELEASE}" CACHE STRING "")
+    set(CMAKE_EXE_LINKER_FLAGS_RELEASE "/DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF ${VCPKG_LINKER_FLAGS} ${VCPKG_LINKER_FLAGS_RELEASE}" CACHE STRING "")
+    set(CMAKE_SHARED_LINKER_FLAGS_DEBUG "${VCPKG_LINKER_FLAGS} ${VCPKG_LINKER_FLAGS_DEBUG}" CACHE STRING "")
+    set(CMAKE_EXE_LINKER_FLAGS_DEBUG "${VCPKG_LINKER_FLAGS} ${VCPKG_LINKER_FLAGS_DEBUG}" CACHE STRING "")
```
Comment (Neumann-A) [inline scripts/toolchains/windows.cmake:36]: That is probably wrong. Use _INIT flags here.

Comment (psfrolov) [reply scripts/toolchains/windows.cmake:36]: All the toolchains in `scripts/toolchains` except Windows consistently use `_INIT` versions of the variables for setting both compiler and linker flags. Should we use `_INIT` variables for all the flags in `windows.cmake` likewise? Or there exists a reason for `windows.cmake` does not use `_INIT`?

**Review by ras0219-msft — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
These regressions will be fixed in another PR.

**Comment by JackBoosY** [comment]
This regression will be solved in #13125.

**Review by JackBoosY — APPROVED** [review]

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Looks like the x64_uwp regressions are sourceforge servers being down

**Comment by BillyONeal** [comment]
Looks like the SourceForge bits are fixed by https://github.com/microsoft/vcpkg/pull/13176

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**